### PR TITLE
KOGITO-1436: Stunner - Remove double `setContent` call. 

### PR DESCRIPTION
**Pull request by ederign**
This PR changes the set content method to return a Promise instead of void.

The goal of this change is to be able to make the loading of editors predictable and remove the need for double set content call.

This PR also includes the changes by @manstis and @romartin  at https://github.com/kiegroup/kie-wb-common/pull/3236/

Merge together with:
kiegroup/kogito-tooling#95
https://github.com/kiegroup/appformer/pull/934

The new changes are here: https://github.com/kiegroup/kie-wb-common/pull/3246/commits/0d5c3a182f32482fa74a58097c72709a57e06171

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-spaces-screen/.gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules/
 dist/
 coverage/
-target/
\ No newline at end of file
+target/
+yarn*.log
```
Comment (ederign) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-spaces-screen/.gitignore]: just a clean up

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/presenters/Viewer.java [hunk 2]
```diff
@@ -40,8 +40,6 @@
         default void onSuccess() {
         }
 
-        ;
-
```
Comment (ederign) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/presenters/Viewer.java]: another clean up

File: kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-standalone/src/main/java/org/kie/workbench/common/stunner/standalone/client/editor/BPMNStandaloneDiagramEditor.java [hunk 3]
```diff
@@ -316,6 +316,7 @@ public void onError(final ClientRuntimeError error) {
                                           BPMNStandaloneDiagramEditor.this.getEditor().onLoadError(error);
                                       }
                                   });
+        return null;
```
Comment (ederign) [inline kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-standalone/src/main/java/org/kie/workbench/common/stunner/standalone/client/editor/BPMNStandaloneDiagramEditor.java]: @romartin I'm not sure what should I return here.

**Comment by romartin** [comment]
Closing as code from this PR is already present in https://github.com/kiegroup/kie-wb-common/pull/3236.

**Comment by romartin** [comment]
@ederign oops sorry closed the wrong PR :/

**Comment by romartin** [comment]
Jenkins execute full downsteam build

**Comment by romartin** [comment]
Hey @ederign @manstis @jomarko @tiagobento 

Great job guys, I've been testing this locally and it results most of the acutal blocker tickets for bpmn editor, although I there is still one concrete case in which it doesn't work properly. 

See the following video:

https://issues.redhat.com/secure/attachment/12468485/KOGITO-1436-verification.mp4

As you can see:
* Now there is no need to click "first" on the canvas, to make properties work, you can just open the editor, update the properties, save and it works (see how the `it-works.bpmn` process is created in the video) - It means removing the double `setContent` calls makes everything work as expected! Great news guys!
* _[BUG]_ On the other hand, I found a bug, that maybe it's still causing confusion to end users, but not related to this. The bug is: after some forms validatoin errors happens, whatever changes to other fields will be not saved properly. So it happens only AFTER some forms validation happens, see how the it-still-does-not-work.bpmn process is being created in the video, but not propertly updated

Sooo IMO we should take these **actions**:

1. Try to merge this PR
2. Try to merge also https://github.com/kiegroup/kie-wb-common/pull/3239
3. Report and fix the issue mentioned above, on top

PS: Also I've been testing this locally with https://github.com/kiegroup/kie-wb-common/pull/3239 merged as well, I mean both features in my local, and Ctrl +S is working fine too now in VSCode, expect in same situation that the bug mentioned above. Good job @Josephblt too!!

Again - great job everyone involed, specially @ederign @tiagobento !!!! :+1: 

CC @LuboTerifaj @domhanak 

Thanks!!
Roger

**Comment by romartin** [comment]
CC @lazarotti @porcelli ^^

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
CC @pefernan maybe I'll need your help Pere, after flushing, once some invalid field, it looks the forms state is not properly saved, see [comment above](https://github.com/kiegroup/kie-wb-common/pull/3246#issuecomment-604209768). If you have some clue anyway please let us know!! :+1:

**Comment by pefernan** [comment]
hey @romartin let me test it locally to understand the issue.

**Comment by domhanak** [comment]
Jenkins execute full downstream build

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/AbstractDMNDiagramEditor.java [hunk 4]
```diff
@@ -430,25 +392,42 @@ public boolean isDirty() {
 
     @Override
     @SetContent
-    public void setContent(final String path,
+    public Promise setContent(final String path,
                            final String value) {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/AbstractDMNDiagramEditor.java]: please update formatting

File: kie-wb-common-kogito/kie-wb-common-kogito-client/src/main/java/org/kie/workbench/common/kogito/client/editor/BaseKogitoEditor.java [hunk 5]
```diff
@@ -145,7 +145,7 @@ public boolean isDirty() {
      * @param value
      * Editor's content
      */
-    public abstract void setContent(final String path, final String value);
+    public abstract Promise setContent(final String path, final String value);
```
Comment (jomarko) [inline kie-wb-common-kogito/kie-wb-common-kogito-client/src/main/java/org/kie/workbench/common/kogito/client/editor/BaseKogitoEditor.java]: In appformer we changed to `Promise<Void>` shouldn't we follow also here?

**Comment by jomarko** [comment]
Manual check in business central was same as on vs code https://github.com/kiegroup/kogito-tooling/pull/95#issuecomment-604468044, but passed all scenarios.

File: kie-wb-common-kogito/kie-wb-common-kogito-client/src/main/java/org/kie/workbench/common/kogito/client/editor/BaseKogitoEditor.java (hunk 5 above)
Comment (tiagobento) [reply kie-wb-common-kogito/kie-wb-common-kogito-client/src/main/java/org/kie/workbench/common/kogito/client/editor/BaseKogitoEditor.java]: That's a current limitation of our annotation processors. Other methods also return untyped Promises for now.

**Comment by romartin** [comment]
Hey @ederign @paulovmr @tiagobento 

Please this PR is blocking this other one for being properly reviewed: https://github.com/kiegroup/kie-wb-common/pull/3239

Could you please prioritize this review and try to merge this ASAP?

Thx!

**Comment by romartin** [comment]
CC @LuboTerifaj @Josephblt ^^

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/AbstractDMNDiagramEditor.java (hunk 4 above)
Comment (ederign) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/editor/AbstractDMNDiagramEditor.java]: fixed

**Pull request merged**